src/storage/store/account/name_registry_events.rs

```diff
@@ -81,6 +82,12 @@ pub fn put_username_proof_transaction(
 
     let secondary_key = make_fname_username_proof_by_fid_key(username_proof.fid);
     txn.put(secondary_key, primary_key);
+
+    // If a username is being transferred, remove the existing secondary key.
+    if let Some(existing_fid) = existing_fid {
+        let secondary_key = make_fname_username_proof_by_fid_key(existing_fid);
+        txn.delete(secondary_key);
```

Comment:
Hmm, have to consider how to handle existing bad data. This doesn't break consensus since it's just an index change. We might need a migration to fix, but we don't have any migration harness code. Will ticket this, let me know if you're interested in taking that on.

Reply:
Feels like it's probably something to deal with outside of this PR